### PR TITLE
[AIRFLOW-6907] Simplify SchedulerJob

### PR DESCRIPTION
**Pull request by mik-laj**
If we realize that some methods always get one constant value, then their logic can be simplified.

---
Issue link: [AIRFLOW-6907](https://issues.apache.org/jira/browse/AIRFLOW-6907)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -1098,37 +1094,20 @@ def _find_executable_task_instances(self, simple_dag_bag, states, session=None):
...
-                and_(DR.dag_id == TI.dag_id, DR.execution_date == TI.execution_date)
+                DR, and_(DR.dag_id == TI.dag_id, DR.execution_date == TI.execution_date)
             )
-            .filter(or_(DR.run_id == None,  # noqa: E711 pylint: disable=singleton-comparison
-                    not_(DR.run_id.like(BackfillJob.ID_PREFIX + '%'))))
+            .filter(or_(DR.run_id.is_(None), not_(DR.run_id.like(BackfillJob.ID_PREFIX + '%'))))
             .outerjoin(DM, DM.dag_id == TI.dag_id)
-            .filter(or_(DM.dag_id == None,  # noqa: E711 pylint: disable=singleton-comparison
-                    not_(DM.is_paused)))
+            .filter(or_(DM.dag_id.is_(None), not_(DM.is_paused)))
```
Comment (ashb) [inline airflow/jobs/scheduler_job.py:1106]: @nuclearpinguin This was an example of where we don't need a noqa anymore -- do you know why you did on your pylint fixes to jobs? Is it because we are ignoring this whole file?

File: airflow/providers/google/cloud/hooks/stackdriver.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 from google.protobuf.json_format import MessageToDict, MessageToJson, Parse
 from googleapiclient.errors import HttpError
 
-from airflow import AirflowException
+from airflow.exceptions import AirflowException
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/stackdriver.py]: Unrelated change here?

Comment (ashb) [reply airflow/providers/google/cloud/hooks/stackdriver.py]: Other than this, :+1: to the change.

File: airflow/jobs/scheduler_job.py (hunk 1 above)
Comment (mik-laj) [reply airflow/jobs/scheduler_job.py:1106]: `DM.dag_id == None` => disable=singleton-comparison
`DM.dag_id.is_(None)` is correct syntax.

File: airflow/providers/google/cloud/hooks/stackdriver.py (hunk 2 above)
Comment (mik-laj) [reply airflow/providers/google/cloud/hooks/stackdriver.py]: This change was already in the master, but the PR preview has not been updated.

File: airflow/jobs/scheduler_job.py (hunk 1 above)
Comment (ashb) [reply airflow/jobs/scheduler_job.py:1106]: Yeah I'm okay with this. My question is why in #7484 does he have

```
models.DagRun.state.is_(None)))  # pylint: disable=no-member
```

but we don't need that here?

Comment (ashb) [reply airflow/jobs/scheduler_job.py:1106]: (Sorry for not being clear. This comment was not about this PR, but about the other one)

**Review by ashb — APPROVED** [review]

File: airflow/jobs/scheduler_job.py (hunk 1 above)
Comment (turbaszek) [reply airflow/jobs/scheduler_job.py:1106]: I don't know, but here we do not run pylint over this file, do we? @ashb

Comment (turbaszek) [reply airflow/jobs/scheduler_job.py:1106]: It's still in TODO list:
https://github.com/PolideaInternal/airflow/blob/37c630da636d1ef352273e33fbdb417727914386/scripts/ci/pylint_todo.txt#L11

And I suppose that once I rebase onto new master I will have to fix it in more places...
